vectortile/src/main/scala/geotrellis/vectortile/protobuf/internal/package.scala

```diff
@@ -0,0 +1,375 @@
+
+  /** If some invalid combination of command id and parameter count are given. */
+  case class CommandError(id: Int, count: Int) extends Exception(s"ID: ${id}, COUNT: ${count}")
+
+  /**
+   * Expand a collection of diffs from some reference point into that
+   * of `Point` values. The default initial reference point is (0,0).
+   */
+  // TODO Use a State Monad to carry the cursor value.
+  private def expand(diffs: Array[(Int, Int)], curs: (Int, Int) = (0, 0)): ListBuffer[(Int, Int)] = {
+    var cursor: (Int, Int) = curs
+    val points = new ListBuffer[(Int, Int)]
+    var i = 0
+
+    diffs.foreach({
+      case (dx, dy) =>
+        val here = (dx + cursor._1, dy + cursor._2)
+
+        points.append(here)
+        cursor = here
+    })
+
+    points
+  }
+
+  /**
+   * Collapse a collection of Points into that of diffs, relative to
+   * the previous point in the sequence.
+   */
+  private def collapse(points: Array[(Int, Int)], curs: (Int, Int) = (0, 0)): Array[(Int, Int)] = {
+    var cursor: (Int, Int) = curs
+    val diffs = new Array[(Int, Int)](points.length)
+    var i = 0
+
+    while (i < points.length) {
+      val curr = points(i)
+      val here = (curr._1 - cursor._1, curr._2 - cursor._2)
+
+      diffs.update(i, here)
+
+      cursor = curr
+
+      i += 1
+    }
+
+    diffs
+  }
+
+  /**
+    * Translate coordinates in VectorTile grid space into real CRS coordinates.
+    *
+    * @param point      A point in a VectorTile geometry, in grid coordinates.
+    * @param topLeft    The location in the current CRS of the top-left corner of this Tile.
+    * @param resolution How much of the CRS's units are covered by a single VT grid coordinate.
+    * @return The [[Point]] projected into the CRS.
+    *
+    * ===Translation Logic===
+    * Extents always exist in ''some'' CRS. Below is information for
+    * determining values related to translating fixed VectorTile grid
+    * coordinates into map coordinates within the implied CRS.
+    *
+    * You can find resolution this way. Let:
+    * {{{
+    * G := Height of grid (# number of cell rows)
+    * T := Height of a Tile (default 4096)
+    * E := Height of the Extent
+    * }}}
+    *
+    * Then the resolution {{{ R = E / (G * T) }}}
+    *
+    * This actually allows same-Tile Layers with different integer `extent` values
+    * to make sense!
+    *
+    * '''Finding the top-left corner'''
+    * {{{
+    * X = xmin + (R * T * SpatialKeyX)
+    * Y = ymax - (R * T * SpatialKeyY)
+    * }}}
+    *
+    * Resolution and the top-left corner only need to be calculated once per Layer.
+    *
+    * '''Shifting the VT grid points'''
+    * {{{
+    * vtX = X + (R * tileX)
+    * vtY = Y - (R * tileY)
+    * }}}
+    *
+    */
+  def toProjection(point: (Int, Int), topLeft: Point, resolution: Double): Point = {
+    Point(
+      topLeft.x + (resolution * point._1),
+      topLeft.y - (resolution * point._2)
+    )
```

Comment:
Do we need to use `Point` here? Would `(Double, Double)` suffice? There is some cost for using `Point`, because it creates a JTS geometry, and if we only need to store x,y double values we should avoid that overhead.